packages/@aws-cdk/aws-lambda/lib/lambda-insights.ts

```diff
@@ -0,0 +1,109 @@
+
+  /**
+   * Version 1.0.89.0
+   */
+  public static readonly VERSION_1_0_89_0 = LambdaInsightsVersion.fromInsightsVersion('1.0.89.0');
+
+  /**
+   * Version 1.0.98.0
+   */
+  public static readonly VERSION_1_0_98_0 = LambdaInsightsVersion.fromInsightsVersion('1.0.98.0');
+
+  /**
+   * Use the insights extension associated with the provided ARN. Make sure the ARN is associated
+   * with same region as your function
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Lambda-Insights-extension-versions.html
+   */
+  public static fromInsightVersionArn(arn: string): LambdaInsightsVersion {
+    return new LambdaInsightsVersion(undefined, arn);
+  }
+
+  // Use the verison to build the object. Not meant to be called by the user -- user should use e.g. VERSION_1_0_54_0
+  private static fromInsightsVersion(insightsVersion: string) {
+    return new LambdaInsightsVersion(insightsVersion, undefined);
+  }
+
+  /**
+   * The arn of the Lambda Insights extension
+   */
+  public readonly layerVersionArn: string;
+
+  private constructor(insightsVersion?: string, arn?: string) {
+    if (arn !== undefined) {
+      // Use arn if explicitly provided
+      this.layerVersionArn = arn;
+    } else if (insightsVersion !== undefined) {
+      // Look up the arn (runs at synthesis time)
+      this.layerVersionArn = Lazy.uncachedString({
+        produce: (context) => this.getVersionArn(context, insightsVersion),
+      });
+    } else {
+      throw new Error('Cannot use Lambda Insights without providing a version or an ARN');
+    }
+  }
+
+  /**
+   * Meant to run at synthesis. It will look up the region in RegionInfo
+   */
+  private getVersionArn(context: any, insightsVersion: string): string {
+    const region = Stack.of(context.scope).region;
+
+    // Check if insights version is valid. This should only happen if one of the public static readonly versions are set incorrectly
+    if (!(insightsVersion in CLOUDWATCH_LAMBDA_INSIGHTS_ARNS)) {
+      throw new Error(`Insights version ${insightsVersion} does not exist. Available versions are ${CLOUDWATCH_LAMBDA_INSIGHTS_ARNS.keys()}`);
+    }
+
+    // Region is defined, look up the arn, or throw an error if the version isn't supported by a region
+    if (region !== undefined && !Token.isUnresolved(region)) {
+      const arn = RegionInfo.get(region).cloudwatchLambdaInsightsArn(insightsVersion);
+      if (arn === undefined) {
+        throw new Error(`Insights version ${insightsVersion} is not supported in region ${region}`);
+      }
+      return arn;
+    }
+    // Otherwise, need to add a mapping to be looked up at deployment time
+    const scopeStack = Stack.of(context.scope);
+
+    // Map name has to include the version, so the primary key becomes the region.
+    // Secondary keys must be alphanumberic. This strategy avoids using secondary keys
+    // because both the version and region are non-alphanumeric
```

Comment:
I don't follow this comment. 

The next line (`mapName`) converts a non-alphanumeric version into an alphanumeric string. So what are we avoiding?

Reply:
I am avoiding this: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/mappings-section-structure.html

`Within a mapping, each map is a key followed by another mapping. The key must be a map of name-value pairs and unique within the mapping. The name can contain only alphanumeric characters (A-Za-z0-9).`

I originally want the mapping to be the following (a single map with all information), but because of the above limitation I couldn't have "1.0.98.0" be a secondary key to the map. Likewise, I couldn't have the region be a secondary key.

```
CloudWatchLambdaInsightsVersions
- us-east-1
- - "1.0.98.0": <arn1>,
- - "1.0.89.0": <arn2>,
- us-east-2
- - "1.0.98.0": <arn3>,
- - "1.0.89.0": <arn4>,
- etc
```

So I devised that the mapping would be the following, which avoids non-alphanumeric characters by just using "arn".

```
CloudWatchLambdaInsightsVersions10980
- us-east-1
- - "arn": <arn1>,
- us-east-2
- - "arn": <arn2>,
- etc
```

If you can think of a more elegant way to do the mapping please let me know, I'd love to improve it.

Reply:
This is fine with me. We can improve this if we find better ways.

If you could update the comment to make this clear in the code, that would be sufficient.

Reply:
Done